cpp/src/parquet/arrow/reader.h

```diff
@@ -126,6 +126,14 @@ class PARQUET_EXPORT FileReader {
                               std::unique_ptr<ParquetFileReader> reader,
                               std::unique_ptr<FileReader>* out);
 
+  /// Factory function to create a FileReader from a ParquetFileReader and properties
+  static ::arrow::Result<std::unique_ptr<FileReader>> Make(
+      ::arrow::MemoryPool* pool, std::unique_ptr<ParquetFileReader> reader,
+      const ArrowReaderProperties& properties);
```

Comment:
Could you add `ARROW_DEPRECATED()` to `Status` versions like https://github.com/apache/arrow/blob/d16ba00a07424ce28eff0c1a4c313c516f5c31bb/cpp/src/parquet/arrow/reader.h#L194-L207 ?

Reply:
"23.0.0". Our next release uses "23.0.0".

> Should I fix this?

Yes.